corehq/apps/hqcase/api/updates.py

```diff
@@ -275,3 +280,57 @@ def _submit_case_blocks(case_blocks, domain, user, device_id):
         device_id=device_id,
         max_wait=15
     )
+
+
+def validate_update_permission(domain, data, user, is_creation):
+    """
+    Check whether the given `user` has permission to create/update a case.
+    Also checks whether `user` has access to all case indices, if there are any.
+    """
```

Comment:
When you say short-circuit if the user is not restricted, do you mean if the user is an admin? I'm not sure how else to go about short-circuiting the function as it needs to verify that the user can access both the `owner_id` and every `case_id` referenced in the case's indices. It will short-circuit however, as soon as it is determined that the user does not have sufficient permission to either of the two previously mentioned properties.

Reply:
yep, that's right